pkg/controllermanager/controller/shoot/shoot_control_delete.go

```diff
@@ -92,23 +92,17 @@ func (c *defaultControl) deleteShoot(o *operation.Operation) *gardenv1beta1.Last
 	// We also check whether the kube-apiserver pod exists in the Shoot namespace within the Seed cluster. If it does not,
 	// then we assume that it has never been deployed successfully. We follow that no resources can have been deployed
 	// at all in the Shoot cluster, thus there is nothing to delete at all.
-	kubeAPIServerFound := false
-	podList, err := botanist.K8sSeedClient.ListPods(o.Shoot.SeedNamespace, metav1.ListOptions{
-		LabelSelector: "app=kubernetes,role=apiserver",
-	})
-	if err != nil {
-		return formatError("Failed to retrieve the list of pods running in the Shoot namespace in the Seed cluster", err)
-	}
-	for _, pod := range podList.Items {
-		if pod.DeletionTimestamp == nil {
-			kubeAPIServerFound = true
-			break
+	noDeletionCheckpointFound := false
+	if _, err := botanist.K8sSeedClient.GetConfigMap(o.Shoot.SeedNamespace, common.DeletionCheckpoint); err != nil {
+		if !apierrors.IsNotFound(err) {
+			return formatError("Failed to retrieve the list of pods running in the Shoot namespace in the Seed cluster", err)
```

Comment:
```suggestion
			return formatError("Failed to retrieve the gardener-shoot-deletion-checkpoint configmap in the Shoot namespace in the Seed cluster", err)
```
Again, you could consider to use `DeletionCheckpont` instead of `gardener-shoot-deletion-checkpoint`.